app.js

```diff
@@ -52,88 +53,84 @@ module.exports = function(someCodeObjects) {
   var server = require('http').createServer(app);
   var io = require('socket.io')(server);
 
-  function getCode(playgroundId, objectId) {
-    if (!codeObjects[playgroundId]) return "";
-    if (!codeObjects[playgroundId][objectId]) return "";
-
-    return codeObjects[playgroundId][objectId].code;
-  }
-
-  function getListOfAllObjects(playgroundId) {
-    var objectIds = Object.keys(codeObjects[playgroundId]);
-    return {objectIds: objectIds};
-  }
-
-  function broadcastObjectList(playgroundId) {
-    io.to(playgroundId).emit('objects list', getListOfAllObjects(playgroundId));
-  }
 
   io.on('connection', function(socket) {
     var query = socket.handshake.query;
-    var playgroundId = query.playgroundId;
+    var playground = world.getOrCreatePlayground(query.playgroundId);
     var client =  query.client;
 
-    socket.join(playgroundId);
-
     if (client == 'renderer')
       rendererUp();
     else
       programmerUp();
 
     function programmerUp() {
-      debug("a new programmer is up for " + playgroundId);
+      debug("a new programmer is up for " + playground.id);
 
-      if (!codeObjects[playgroundId]) return;
-      socket.emit('objects list', getListOfAllObjects(playgroundId));
+      if (!playground.isEmpty())
+        sendListOfAllOBjects();
     }
 
+
     function rendererUp() {
-      debug("a new renderer is up for " + playgroundId);
+      debug("a new renderer is up for " + playground.id);
 
-      if (!codeObjects[playgroundId]) return;
-      socket.emit('playground full update', codeObjects[playgroundId]);
-    }
+      if (playground.isEmpty()) return;
 
-    socket.on('code update', function(data) {
-      var objectId = data.objectId;
+      var data = Object.create(null);
+      playground.population().forEach(function(codeObjectId) {
+        data[codeObjectId] = playground.getOrCreateCodeObject(codeObjectId).getData();
+      });
```

Comment:
In my view, the argument for keeping this `data` construction business (essentially serializing the playground for transport) outside of the `Playground` class is that we are keeping serialization a separate concern from the models themselves.

But since `CodeObject` ends up violating this separation by exposing a `getData` method, why not go all the way and move the code above into a `Playground.getData` method?